distributed/worker.py

```diff
@@ -2780,6 +2851,10 @@ async def gather_dep(
                     import pdb
 
                     pdb.set_trace()
+                msg = error_message(e)
+                for k in self.in_flight_workers[worker]:
+                    ts = self.tasks[k]
+                    recommendations[ts] = tuple(msg.values())
                 raise
```

Comment:
This unhandled exception was also capable of causing a deadlock. That's also fixed now